tests/Feature/Api/Collections/IndexTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Tests\Feature\Api\Collections;
+
+use App\Collection;
+use App\Elasticsearch\Repositories\ItemRepository;
+use App\Item;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Http\Client\Request;
+use Illuminate\Support\Facades\Http;
+use Tests\RecreateSearchIndex;
+use Tests\TestCase;
+
+class IndexTest extends TestCase
+{
+    use RefreshDatabase;
+    use RecreateSearchIndex;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        Http::fake([
+            'http://localhost/api/v1/items*' => function (Request $request) {
+                $testResponse = $this->get($request->url());
+                return Http::response(
+                    $testResponse->getContent(),
+                    $testResponse->getStatusCode(),
+                    $testResponse->headers->all()
+                );
+            },
+        ]);
+    }
+
+    public function testWithUrl()
+    {
+        Item::factory()->create(['author' => 'author-1']);
+        Item::factory()->create(['author' => 'author-2']);
+        $collection = Collection::factory()
+            ->published()
+            ->create([
+                'url' => 'https://www.webumenia.sk/katalog-new?filter[author][]=author-1',
+            ]);
+
+        // Ensure all data is indexed
+        app(ItemRepository::class)->refreshIndex();
+
+        $response = $this->get(route('api.collections.index'));
+        $response->assertJsonCount(1, 'data')->assertJsonPath('data.0', [
+            'id' => $collection->id,
+            'name' => $collection->name,
+            'text' => $collection->text,
+            'filter_items_url' => route('api.v1.items.index', [
+                'filter' => [
+                    'author' => ['author-1'],
+                ],
+            ]),
+            'filter_items_count' => 1,
```

Comment:
IMHO I think the ideal API would just give me
```suggestion
            'id' => $collection->id,
            'name' => $collection->name,
            'text' => $collection->text,
            'items' => [$item1, $item2] // items list/resouce,
```
And I shouldn't need to care if the items are coming from URL or a list of IDs.

Reply:
It would be better to retrieve paginated items from a separate endpoint, such as `api/collection/{id}/items`. However, at this point, we only need to implement filtered collections for MG. Therefore, the existing endpoint `api/v1/items` could suffice.